agent/battle/vision/provider.py

```
@@ -0,0 +1,111 @@
+    if not config.endpoint:
+        raise ValueError("vision endpoint is required when enabled")
+    if not config.model:
+        raise ValueError("vision model is required when enabled")
+    api_key = config.api_key or os.environ.get("MAAFGO_VISION_API_KEY", "local")
+    return OpenAICompatibleVisionProvider(config.endpoint, api_key, config.model, timeout_s=config.timeout_s)
+
+
+def _extract_content(data: dict[str, Any]) -> str:
+    choices = data.get("choices")
+    if not isinstance(choices, list) or not choices:
+        raise ValueError("response choices is empty")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)